ddtrace/appsec/_iast/_metrics.py

```diff
@@ -98,10 +102,48 @@ def _set_metric_iast_executed_sink(vulnerability_type):
     )
 
 
-@metric_verbosity(TELEMETRY_INFORMATION_VERBOSITY)
-def _set_metric_iast_request_tainted():
+def _request_tainted():
     from ._taint_tracking import num_objects_tainted
 
-    total_objects_tainted = num_objects_tainted()
+    return num_objects_tainted()
+
+
+@metric_verbosity(TELEMETRY_INFORMATION_VERBOSITY)
+def _set_metric_iast_request_tainted():
+    total_objects_tainted = _request_tainted()
     if total_objects_tainted > 0:
         telemetry_writer.add_count_metric(TELEMETRY_NAMESPACE_TAG_IAST, "request.tainted", total_objects_tainted)
+
+
+def _set_span_tag_iast_request_tainted(span):
+    total_objects_tainted = _request_tainted()
+
+    if total_objects_tainted > 0:
+        span.set_tag(IAST_SPAN_TAGS.TELEMETRY_REQUEST_TAINTED, total_objects_tainted)
+
+
+def _set_span_tag_iast_executed_sink(span):
+    data = get_iast_span_metrics()
+
+    if data is not None:
+        for key, value in data.items():
+            if key.startswith(IAST_SPAN_TAGS.TELEMETRY_EXECUTED_SINK):
+                span.set_tag(key, value)
+
+    reset_iast_span_metrics()
+
+
+def increment_iast_span_metric(prefix: str, metric_key: str, counter: int = 1) -> None:
+    data = get_iast_span_metrics()
+    full_key = prefix + "." + metric_key.lower()
+    result = data.get(full_key, 0)
+    data[full_key] = result + counter
+
+
+def get_iast_span_metrics() -> Dict:
+    return _IAST_SPAN_METRICS
+
+
+def reset_iast_span_metrics() -> None:
+    global _IAST_SPAN_METRICS
+    _IAST_SPAN_METRICS = dict()
```

Comment:
```suggestion
    _IAST_SPAN_METRICS.clear()
```
It doesn't create an additional object/reference and release the memory immediatly